src/main/resources/static/js/mods.js

```diff
@@ -34,7 +47,8 @@ angular.module('Mccy.mods',[
             restrict: 'A',
             scope: {
                 mod: '=mccyModPanel',
-                showDelete: '=mccyModPanelShowDelete'
+                showDelete: '=mccyModPanelShowDelete',
+                versions: '=mccyModPanelVersions'
```

Comment:
Since you don't need `$scope.versions` from within your directive, just remove this entry.